File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/customobjectbulkjobresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class CustomObjectBulkJobResponseError extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code?: number;

  @SpeakeasyMetadata({ data: "json, name=message" })
  message?: string;
}

export enum CustomObjectBulkJobResponseNamespaceEnum {
    Default = "default",
    ComZuora = "com_zuora"
}

export enum CustomObjectBulkJobResponseOperationEnum {
    Delete = "delete",
    Create = "create"
}

export enum CustomObjectBulkJobResponseStatusEnum {
    Open = "open",
    Pending = "pending",
    InProgress = "in_progress",
    Completed = "completed",
    Failed = "failed",
    Cancelled = "cancelled"
}


export class CustomObjectBulkJobResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=CreatedById" })
  createdById?: string;

  @SpeakeasyMetadata({ data: "json, name=CreatedDate" })
  createdDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=Id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=UpdatedById" })
  updatedById?: string;

  @SpeakeasyMetadata({ data: "json, name=UpdatedDate" })
  updatedDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=error" })
  error?: CustomObjectBulkJobResponseError;

  @SpeakeasyMetadata({ data: "json, name=namespace" })
  namespace?: CustomObjectBulkJobResponseNamespaceEnum;

  @SpeakeasyMetadata({ data: "json, name=object" })
  object?: string;

  @SpeakeasyMetadata({ data: "json, name=operation" })
  operation?: CustomObjectBulkJobResponseOperationEnum;

  @SpeakeasyMetadata({ data: "json, name=processingTime" })
  processingTime?: number;

  @SpeakeasyMetadata({ data: "json, name=recordsProcessed" })
  recordsProcessed?: number;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: CustomObjectBulkJobResponseStatusEnum;
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/shared/customobjectbulkjobresponse.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class CustomObjectBulkJobResponseError:
    code: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    
class CustomObjectBulkJobResponseNamespaceEnum(str, Enum):
    DEFAULT = "default"
    COM_ZUORA = "com_zuora"

class CustomObjectBulkJobResponseOperationEnum(str, Enum):
    DELETE = "delete"
    CREATE = "create"

class CustomObjectBulkJobResponseStatusEnum(str, Enum):
    OPEN = "open"
    PENDING = "pending"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


@dataclass_json
@dataclasses.dataclass
class CustomObjectBulkJobResponse:
    created_by_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CreatedById') }})
    created_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CreatedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Id') }})
    updated_by_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('UpdatedById') }})
    updated_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('UpdatedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    error: Optional[CustomObjectBulkJobResponseError] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('error') }})
    namespace: Optional[CustomObjectBulkJobResponseNamespaceEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('namespace') }})
    object: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('object') }})
    operation: Optional[CustomObjectBulkJobResponseOperationEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('operation') }})
    processing_time: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('processingTime') }})
    records_processed: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('recordsProcessed') }})
    status: Optional[CustomObjectBulkJobResponseStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    
